README.md

```diff
@@ -1,4 +1,208 @@
+
+Let's add an side effect that makes a network request each time when action is `CalculatorAction.Add`
+and current state is `9`. If server responds with http code `200` - side effect should emit `CalculatorAction.Deduct(1)`.
+```kotlin
+val sideEffect = object : SideEffect<Int, CalculatorAction> {
+    override val name: String = "network logger"
+
+    override fun CoroutineScope.start(
+        input: ReceiveChannel<CalculatorAction>,
+        stateAccessor: StateAccessor<Int>,
+        output: SendChannel<CalculatorAction>,
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() == 9) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Now we need to modify how store is created by adding this `sideEffect`:
+```kotlin
+val storeWithSideEffect = GlobalScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions - 
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Add(9)) 
+// Current state will become 9
+storeWithSideEffect.dispatch(CalculatorAction.Add(1)) 
+// Current state will become 10 and network request will happen
+```
+
+Library provides useful simplified versions of `SideEffect` interface:
+- `SimpleSideEffect` that produces on one input action either one output action or no action.
+- `CancellableSideEffect` that cancels currently executed `Job` and starts a new one.
+
+### Implementation details
+
+By default, CoRedux starts main reducer coroutine ("manager"), when `StateReceiver` subscribes to `CoreduxStore` instance.
```

Comment:
I would remove the `By default` as again, is it possible to change that? You just can change the default coroutine the `reducer` runs, right? but you can't change the overall behavior that it runs on a "reducer coroutine"

Reply:
> is it possible to change that

Yes, I've answered about it in some previous reply.

User can define whether reducer coroutine start immediately (so it is possible to dispatch action and update the state) or when first `StateReceiver` subscribes.